Bugfix [PDF Refactor] FXIOS Disable share button for toolbar when loading website

FilippoZazzeroni opened the pull request:
## :scroll: Tickets
### Large PDF can be shared while downloading
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-11711)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/25523)

### Weird display after scenario
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-11515)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/25071)

## :bulb: Description
Disable share button on Toolbar while loading any url. Cancel download when tab is deleted.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [x] If needed, I updated documentation / comments for complex code and public methods
- [x] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

FilippoZazzeroni commented:
@mergifyio backport release/v138

FilippoZazzeroni commented on firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+WebViewDelegates.swift:
```diff
@@ -773,26 +771,26 @@ extension BrowserViewController: WKNavigationDelegate {
             tempPDF.onDownloadError = { error in
-                self?.navigationHandler?.removeDocumentLoading()
-                guard let error, let webView else { return }
+                self?.navigationHandler?.removeDocumentLoading(completion: nil)
```
Yes right, we are using it only for the tests at the moment.

cyndichin reviewed (commented):
thanks for writing tests @FilippoZazzeroni. I only reviewed the unit tests, looks good. Added some thoughts, which can def be done in a followup PR to avoid blocking this one.

FilippoZazzeroni commented on firefox-ios/firefox-ios-tests/Tests/ClientTests/Coordinators/Mocks/MockBrowserCoordinator.swift:
```diff
@@ -167,7 +167,7 @@ class MockBrowserCoordinator: BrowserNavigationHandler, ParentCoordinatorDelegat
     }
 
-    func removeDocumentLoading() {
+    func removeDocumentLoading(completion: (() -> Void)?) {
```
I agree @cyndichin me and @lmarceau added this to fix a flaky test, let me think how can i fix this.

cyndichin replied:
ah ok let me know if its still flaky, do you have the PR for that? I can also take a look at it during the West Office Hours.

FilippoZazzeroni replied:
yes sure this was the [PR](https://github.com/mozilla-mobile/firefox-ios/pull/24904).

FilippoZazzeroni commented:
@thatswinnie i had to merge `AddressBarState` with main and custom resolve some conflicts, please can you check that is everything as expected ? @cyndichin i've removed the `completionHandler` from `removeDocumentLoading` and updated also a bit the structure of `BrowserCoordinatorTests`
